dspace-server-webapp/src/test/java/org/dspace/app/rest/SubscriptionRestRepositoryIT.java

```diff
@@ -0,0 +1,847 @@
+                                 .content(new ObjectMapper().writeValueAsString(map))
+                                 .contentType(MediaType.APPLICATION_JSON_VALUE))
+                       .andExpect(status().isCreated())
+                       .andExpect(jsonPath("$.subscriptionType", is("testType")))
+                       .andExpect(jsonPath("$.subscriptionParameterList[0].name", is("Frequency")))
+                       .andExpect(jsonPath("$.subscriptionParameterList[0].value", is("daily")))
+                       .andExpect(jsonPath("$._links.ePerson.href", Matchers.endsWith("ePerson")))
+                       .andExpect(jsonPath("$._links.dSpaceObject.href", Matchers.endsWith("dSpaceObject")))
+                       .andDo(result -> idRef.set(read(result.getResponse().getContentAsString(), "$.id")));
+        } finally {
+            SubscribeBuilder.deleteSubscription(idRef.get());
+        }
+    }
+
+    @Test
+    public void createSubscriptionByEPersonTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        Map<String, Object> map = new HashMap<>();
+        map.put("subscriptionType", "testType");
+        List<Map<String, Object>> list = new ArrayList<>();
+        Map<String, Object> sub_list = new HashMap<>();
+        sub_list.put("name", "Frequency");
+        sub_list.put("value", "daily");
+        list.add(sub_list);
+        map.put("subscriptionParameterList", list);
+
+        context.restoreAuthSystemState();
+
+        AtomicReference<Integer> idRef = new AtomicReference<Integer>();
+
+        try {
+            String tokenEPerson = getAuthToken(eperson.getEmail(), password);
+            getClient(tokenEPerson).perform(post("/api/core/subscriptions")
+                                   .param("dspace_object_id", publicItem.getID().toString())
+                                   .param("eperson_id", eperson.getID().toString())
+                                   .content(new ObjectMapper().writeValueAsString(map))
+                                   .contentType(MediaType.APPLICATION_JSON_VALUE))
+                       .andExpect(status().isCreated())
+                       .andExpect(jsonPath("$.subscriptionType", is("testType")))
+                       .andExpect(jsonPath("$.subscriptionParameterList[0].name", is("Frequency")))
+                       .andExpect(jsonPath("$.subscriptionParameterList[0].value", is("daily")))
+                       .andExpect(jsonPath("$._links.ePerson.href", Matchers.endsWith("ePerson")))
+                       .andExpect(jsonPath("$._links.dSpaceObject.href", Matchers.endsWith("dSpaceObject")))
+                       .andDo(result -> idRef.set(read(result.getResponse().getContentAsString(), "$.id")));
+        } finally {
+            SubscribeBuilder.deleteSubscription(idRef.get());
+        }
+    }
+
+    @Test
+    public void createInvalidSubscriptionTest() throws Exception {
```

Comment:
It's unclear to me why the subscription is invalid here?  What makes it invalid?

On that note, why aren't their any tests which add paramaters that are *not* "Frequency"?  Is "Frequency" the only valid parameter?  What happens if I set "Frequency = nothing"?  Is that valid?